snap7/util.py

```diff
@@ -619,20 +624,24 @@ def get_value(self, byte_index, _type):
 
         raise ValueError
 
-    def set_value(self, byte_index, _type, value):
+    def set_value(self, byte_index: Union[str, int], _type: str, value):
         _bytearray = self.get_bytearray()
 
         if _type == 'BOOL':
-            byte_index, bool_index = byte_index.split('.')
-            return set_bool(_bytearray, self.get_offset(byte_index),
-                            int(bool_index), value)
+            if isinstance(byte_index, str):
+                byte_index, bool_index = byte_index.split('.')
+                return set_bool(_bytearray, self.get_offset(byte_index),
+                                int(bool_index), value)
 
         byte_index = self.get_offset(byte_index)
 
         if _type.startswith('STRING'):
-            max_size = re.search(r'\d+', _type).group(0)
-            max_size = int(max_size)
-            return set_string(_bytearray, byte_index, value, max_size)
+            max_size = re.search(r'\d+', _type)
+            if max_size is None:
+                raise Snap7Exception("Max size could not be determinate. re.search() returned None")
```

Comment:
It makes prs easier to review and quicker to process if you keep them single purpose

Reply:
The problem here was, that mypy complained about overloading and incompatible types.
One is, that re.search returns an Optional[Match[str]] type. So max_size is either Match[str] or None. So both cases have to be cought, otherwise mypy complains.
Then with .group(0) you would change type from Match[str] to AnyStr.
And in the next line again from AnyStr to int.
I agree, that the way before was more elegant (if you know what you do), but otherwise I would have to find a way to suppress it. 😕